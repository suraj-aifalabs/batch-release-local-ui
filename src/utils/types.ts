export interface TrackingRow {
  _id: string;
  stage: string;
  status: string;
  cequenceId_Hashed?: string;
  cquenceDIN: string;
  cequenceId?: string;
  patientName: string;
  patientDOB: string;
  patientWeight: string;
  cquenceOrderId: string;
  country: string;
  coicBagId: string;
  batchNumber: string;
  createdAt: string;
  updatedAt: string;
  __v: number;
  pccNumber: string;
  totalVolume: string;
  productDose: string;
  marketAuthorizationNumber: string;
  nameAndAddress: string;
  productNDC: string;
  expirationDate: string;
}
