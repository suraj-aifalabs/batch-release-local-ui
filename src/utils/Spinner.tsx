export function Spinner() {
    return (
        <div>...</div>
    );
}