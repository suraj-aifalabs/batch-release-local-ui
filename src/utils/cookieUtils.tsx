import Cookies from "universal-cookie";
const cookies = new Cookies();

const SECURE_IDS = {
    msal_id_token: "91b84dd50961309e2720e24903e1e6cd",
    msal_access_token: "d95ec1a6c067bdc116d48fcc923c7c75",
    username: "368bf182bbac3104e48cbe3213c3af50",
    user_fullname: "10ec2c8cd682a81da2adff0e0cc9660a",
    browser_id: "eab1a3a68fd07cfadb5f81da550b567b"
} as const;

export const getFullNameFromCookie = () => cookies.get(SECURE_IDS["user_fullname"]);

export const getUsernameFromCookie = () => cookies.get(SECURE_IDS["username"]);

export const getAccessTokenFromCookie = () => cookies.get(SECURE_IDS["msal_access_token"]);

export const getIdTokenFromCookie = () => cookies.get(SECURE_IDS["msal_id_token"]);

export const getBrowserIdFromCookie = () => cookies.get(SECURE_IDS["browser_id"]);
