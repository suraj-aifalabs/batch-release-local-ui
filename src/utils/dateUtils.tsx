export const formatDateTime = (dateString: string | null | undefined): string => {
    if (!dateString) {
        return "N/A";
    }

    const date = new Date(dateString);

    // Check if the date is invalid
    if (isNaN(date.getTime())) {
        return "Invalid Date";
    }

    const dateOptions: Intl.DateTimeFormatOptions = {
        day: "2-digit",
        month: "short",
        year: "numeric",
        timeZone: Intl.DateTimeFormat().resolvedOptions().timeZone,
    };

    const timeOptions: Intl.DateTimeFormatOptions = {
        hour: "2-digit",
        minute: "2-digit",
        hour12: true,
        timeZone: Intl.DateTimeFormat().resolvedOptions().timeZone,
    };

    // Format date and time separately, then combine
    const formattedDate = date.toLocaleDateString("en-GB", dateOptions).replace(/\s/g, "-");
    const formattedTime = date.toLocaleTimeString("en-GB", timeOptions);

    return `${formattedDate} ${formattedTime}`;
};