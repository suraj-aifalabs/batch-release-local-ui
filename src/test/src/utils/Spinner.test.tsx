import { render, screen } from '@testing-library/react';
import { Spinner } from '@/utils/Spinner';

describe('Spinner', () => {
    it('renders the Spinner component with correct content', () => {
        render(<Spinner />);
        const spinnerElement = screen.getByText('...');
        expect(spinnerElement).toBeInTheDocument();
    });
});
