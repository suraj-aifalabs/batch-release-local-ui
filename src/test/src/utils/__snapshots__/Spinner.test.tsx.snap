// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Component should match snapshot 1`] = `
<DocumentFragment>
  <div>
    ...
  </div>
</DocumentFragment>
`;
